### PR TITLE
#182 CosmosDb Support

### LINKED ISSUE #182: CosmosDb Support
As an engineer, I would like to create a function/cmdlet to support CosmosDb checks so that we can support future customer's needs.

- [ ] Create `Confirm-CosmosDb.ps1` and unit tests
- [ ] Wire up `Confirm-CosmosDb` to `Common.ps1`
- [ ] Create example tests, bicep and documentation in `examples` folder

### PR DESCRIPTION
**Pull request by marclerwick**
Added support for Cosmos DB Account, Cosmos DB Gremlin Database, Cosmos DB Mongo DB Database, and Cosmos DB SQL Database.

Closes #182

### CODE REVIEW

**Comment by marclerwick** [comment]
Created as a draft. I still have to confirm the bicep file and the example tests work as expected. I'll be out for a long weekend and wanted to get feedback while I'm gone.

File: Modules/BenchPress.Azure/Public/Confirm-CosmosDBGremlinDatabase.ps1 [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+      The name of the Resource Group.
+
+    .PARAMETER AccountName
+      The Cosmos DB account name.
+
+    .PARAMETER Name
+      The name of the Cosmos DB Gremlin Database
+
+    .EXAMPLE
+      Confirm-AzBPCosmosDBGremlinDatabase -Name "benchpresstest" -ResourceGroupName "rgbenchpresstest" -DatabaseName "db"
```
Comment (msha1026) [inline Modules/BenchPress.Azure/Public/Confirm-CosmosDBGremlinDatabase.ps1:26]: `-DatabaseName "db"` -> `AccountName "an"`

File: Modules/BenchPress.Azure/Public/Confirm-CosmosDBMongoDBDatabase.ps1 [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+      The name of the Resource Group.
+
+    .PARAMETER AccountName
+      The Cosmos DB account name.
+
+    .PARAMETER Name
+      The name of the Cosmos DB Mongo DB Database
+
+    .EXAMPLE
+      Confirm-AzBPCosmosDBMongoDBDatabase -Name "benchpresstest" -ResourceGroupName "rgbenchpresstest" -DatabaseName "db"
```
Comment (msha1026) [inline Modules/BenchPress.Azure/Public/Confirm-CosmosDBMongoDBDatabase.ps1:26]: `-DatabaseName "db"` -> `AccountName "an"`

File: Modules/BenchPress.Azure/Public/Confirm-CosmosDBSqlDatabase.ps1 [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+      The name of the Resource Group.
+
+    .PARAMETER AccountName
+      The Cosmos DB account name.
+
+    .PARAMETER Name
+      The name of the Cosmos DB SQL Database
+
+    .EXAMPLE
+      Confirm-AzBPCosmosDBSqlDatabase -Name "benchpresstest" -ResourceGroupName "rgbenchpresstest" -DatabaseName "db"
```
Comment (msha1026) [inline Modules/BenchPress.Azure/Public/Confirm-CosmosDBSqlDatabase.ps1:26]: `-DatabaseName "db"` -> `AccountName "an"`

File: Modules/BenchPress.Azure/Public/Get-ResourceByType.ps1 [hunk 4]
```diff
@@ -81,16 +84,19 @@ function Get-ResourceByType {
     [string]$ResourceGroupName,
 
     [Parameter(Mandatory = $true)]
-    [ValidateSet("ActionGroup", "AksCluster", "AppInsights", "AppServicePlan", "ContainerRegistry", "KeyVault",
-    "OperationalInsightsWorkspace", "ResourceGroup", "SqlDatabase", "SqlServer", "StorageAccount",
-    "SynapseSparkPool", "SynapseSqlPool", "SynapseWorkspace", "VirtualMachine", "WebApp")]
+    [ValidateSet("ActionGroup", "AksCluster", "AppInsights", "AppServicePlan", "ContainerRegistry", "CosmosDBAccount",
+    "CosmosDBSqlDatabase", "KeyVault", "OperationalInsightsWorkspace", "ResourceGroup", "SqlDatabase", "SqlServer",
```
Comment (msha1026) [inline Modules/BenchPress.Azure/Public/Get-ResourceByType.ps1:88]: Missing Gremlin and Mongo

**Review by msha1026 — APPROVED** [review]
Looks good other than the small fixes!

File: Modules/BenchPress.Azure/Public/Confirm-CosmosDBGremlinDatabase.ps1 (hunk 1 above)
Comment (marclerwick) [reply Modules/BenchPress.Azure/Public/Confirm-CosmosDBGremlinDatabase.ps1]: Good catches! Fixed them all.

File: Modules/BenchPress.Azure/Public/Get-ResourceByType.ps1 (hunk 4 above)
Comment (marclerwick) [reply Modules/BenchPress.Azure/Public/Get-ResourceByType.ps1]: Fixed, thanks!

**Comment by marclerwick** [comment]
Ready for review.

**Review by ozboms — APPROVED** [review]

File: examples/CosmosDB/CosmosDB.Tests.ps1 [hunk 5]
```diff
@@ -0,0 +1,391 @@
...
+    $result.Success | Should -Be $true
+  }
+
+  It 'Should not contain a Cosmos DB account with the given name' {
+    #arrange
+    # The 'ErrorAction = SilentlyContinue' command suppresses all errors.
+    # In this test, it will suppress the error message when a resource cannot be found.
+    # Remove this field to see all errors.
+    $params = @{
+      ResourceGroupName = "rg-name"
```
[the remark this answers is not in the document]
Comment (marclerwick) [reply examples/CosmosDB/CosmosDB.Tests.ps1]: Good catch! Fixed.

**Pull request merged**